src/config.rs

```diff
@@ -170,17 +170,18 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
     } else {
         "https://localhost:15012".to_string()
     };
-
-    let xds_address = parse(XDS_ADDRESS)?
-        .or(pc.discovery_address)
-        .or_else(|| Some(default_istiod_address.clone()));
+    let xds_address = empty_to_none(
+        parse(XDS_ADDRESS)?
```

Comment:
why not fix the parse and parse default?

Reply:
parse distinguishes between unset (None) and set to "" (Some("")). I don't think that's a bug, just different use case

Reply:
But we do not need to distinguish unset and empty "".

Reply:
XDS is disabled by setting to "". Unset means to use the default address